common/rpc/interceptor/namespace_validator.go

```diff
@@ -128,12 +137,16 @@ func (ni *NamespaceValidatorInterceptor) NamespaceValidateIntercept(
 	info *grpc.UnaryServerInfo,
 	handler grpc.UnaryHandler,
 ) (interface{}, error) {
-	err := ni.setNamespaceIfNotPresent(req)
-	if err != nil {
-		return nil, err
-	}
 	reqWithNamespace, hasNamespace := req.(NamespaceNameGetter)
 	if hasNamespace {
+
+		if reqWithNamespace.GetNamespace() == "" {
+			err := errNamespaceNotSet
+			metrics.ServiceErrNamespaceNotSetCounter.With(ni.metricsHandler).Record(1)
```

Comment:
Tag the metric will operation name will be useful I think.

Reply:
Thanks. I wasn't sure what tags would make sense here. Its not easy to verify which interceptors are used by which service.